1.For the titanic dataset perform the hypothesis testing to claim that there is a
significant difference between  proportions of survived females to that of survived
males.

2.For the iris dataset perform the ANOVA testing.
“Normally, species variable put as dependent variable to do classification. But species also can put as independent variable to conduct ANOVA test. Conduct the statistical test that figures out three or more differences in group means.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats 

1.For the titanic dataset perform the hypothesis testing to claim that there is a significant difference between proportions of survived females to that of survived males.

In [2]:
df=pd.read_csv("titanic.csv")
df.head(2)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked  
0      0  A/5 21171   7.2500   NaN        S  
1      0   PC 17599  71.2833   C85        C

In [3]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [5]:
# No null values in PassengerId and Survived columns, so we can use these columns for calculations.

In [6]:
total_passengers=df["PassengerId"].count()
print(f"total_passengers: {total_passengers}")

total_passengers: 891


In [7]:
survived=df.groupby("Sex")["Survived"].sum()
print(survived)
survived_total=df.groupby("Sex")["Survived"].sum().sum()
print(f"survived_total: {survived_total}")

Sex
female    233
male      109
Name: Survived, dtype: int64
survived_total: 342


In [8]:
df_female=df[df["Sex"]=="female"][df["Survived"]==1]
female_survived=df_female.Sex.count()
print(f"female_survived= {female_survived}")

female_survived= 233


C:\Users\om\AppData\Local\Temp\ipykernel_16004\2117234647.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_female=df[df["Sex"]=="female"][df["Survived"]==1]


In [9]:
df_male=df[df["Sex"]=="male"][df["Survived"]==1]
male_survived=df_male.Sex.count()
print(f"male_survived= {male_survived}")

male_survived= 109


C:\Users\om\AppData\Local\Temp\ipykernel_16004\2365788845.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_male=df[df["Sex"]=="male"][df["Survived"]==1]


In [10]:
male_survived_proportion=male_survived/total_passengers
print(f"male_survived_proportion= {male_survived_proportion}")
female_survived_proportion=female_survived/total_passengers
print(f"female_survived_proportion= {female_survived_proportion}")
total_survived_proportion=survived_total/total_passengers
print(f"total_survived_proportion= {total_survived_proportion}")

male_survived_proportion= 0.122334455667789
female_survived_proportion= 0.2615039281705948
total_survived_proportion= 0.3838383838383838


In [11]:
p1=female_survived_proportion
p2=male_survived_proportion
n1=female_survived
n2=male_survived
p=total_survived_proportion
nullHypothesisH0="nodifference" #p1=p2
alternativeHypothesis="difference"# p1 != p2
z=(p1-p2)/(p*(1-p)*((1/n1)+(1/n2)))**0.5
z                   

2.4660428371173393

In [12]:
print("z > 1.96 i.e., 2.46 so we reject the H0 , there is a significant difference between proportions of survived females to that of survived males.")

z > 1.96 i.e., 2.46 so we reject the H0 , there is a significant difference between proportions of survived females to that of survived males.


2.For the iris dataset perform the ANOVA testing.
“Normally, species variable put as dependent variable to do classification. But species also can put as independent variable to conduct ANOVA test. Conduct the statistical test that figures out three or more differences in group means.

In [13]:
df_iris=pd.read_csv("iris.csv")
df_iris.head(2)

sepal.length  sepal.width  petal.length  petal.width variety
0           5.1          3.5           1.4          0.2  Setosa
1           4.9          3.0           1.4          0.2  Setosa

In [14]:
print(df_iris.shape)

(150, 5)


One-Way ANOVA in Python: One-way ANOVA (also known as “analysis of variance”) is a test that is used to find out whether there exists a statistically significant difference between the mean values of more than one group.
Hypothesis involved:

A one-way ANOVA has the below given null and alternative hypotheses:

    H0 (null hypothesis): μ1 = μ2 = μ3 = … = μk (It implies that the means of all the population are equal)
    H1 (null hypothesis): It states that there will be at least one population mean that differs from the rest


In [15]:
from scipy.stats import f_oneway
p1=list(df_iris["sepal.length"])
p2=list(df_iris["sepal.width"])
p3=list(df_iris["petal.length"])
p4=list(df_iris["petal.width"])
f_oneway(p1,p2,p3,p4)

F_onewayResult(statistic=482.91531656927964, pvalue=4.660592480454751e-159)

Analyse the result:
The F statistic and p-value turn out to be equal to 482.915, pvalue=4.66e-159 respectively. Since the p-value is less than 0.05 hence we would reject the null hypothesis.
Conclude that there is a significance difference between means of different features.